### PR TITLE
feat: Draft production warning banner

### LINKED ISSUE #283: Draft Production Warning Banner
When a user is browsing a public facing production page (e.g. the production view page or the booking page for the production), we should show a banner at the top to tell the user that this page is private as the production is not public yet.

Something like this
image

### PR DESCRIPTION
**Pull request by streth11**
closes #283

### CODE REVIEW

File: components/production/ProductionDraftWarningBanner.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+<script setup lang="ts">
+import ProductionStatusEnum from '~~/enums/ProductionStatusEnum';
+
+const props = defineProps({
+  production: {
+    required: true,
+    type: Object
+  }
+});
```
Comment (atoff) [inline components/production/ProductionDraftWarningBanner.vue:18]: With typescript, there is a better way to define this (currently "production" just has to be an object, which isn't very useful for making sure it is passed the right stuff (what if status isn't defined in it??)) - https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

Something like this would be better and type-safe:
```ts suggestion
import {ProductionStatus} from '~/graphql/codegen/operations'

const props = defineProps<{
    production: {status: ProductionStatus}
}>();
```

This would make sure the object given as a prop is a object which contains a `status` key, where the value of the key in one of the possible `ProductionStatus` enums

**Review by atoff — CHANGES_REQUESTED** [review]

**Comment by atoff** [comment]
Also can you add a test for this? Can be quite simple, but at a minimum:
- In the production page test, ensure the component exists (https://github.com/BristolSTA/uobtheatre-web/blob/develop/tests/unit/pages/production/Production.spec.js)
- Make a new unit test in `components` for the new component, and make sure it shows when in draft, but don't when published

**Review by atoff — APPROVED** [review]
looks good! 🚀
